distributed/deploy/adaptive.py

```diff
@@ -104,12 +110,48 @@ def __init__(
             target_duration = dask.config.get("distributed.adaptive.target-duration")
 
         self.target_duration = parse_timedelta(target_duration)
+        self.interval = parse_timedelta(interval, "seconds")
+        self.wait_count = wait_count
+        self.minimum = minimum
+        self.maximum = maximum
+        self.periodic_callback = None
+
+        def f():
+            try:
+                self.periodic_callback.start()
+            except AttributeError:
+                pass
+
+        if self.interval:
+            import weakref
+
+            self_ref = weakref.ref(self)
+
+            async def _adapt():
+                core = self_ref()
+                if core:
+                    await core.adapt()
+
+            self.periodic_callback = PeriodicCallback(_adapt, self.interval * 1000)
+            try:
+                self.loop.add_callback(f)
+            except AttributeError:
+                IOLoop.current().add_callback(f)
+
+        try:
+            self.plan = set()
+            self.requested = set()
+            self.observed = set()
+        except Exception:
+            pass
```

Comment:
This will always end up in an exception since below these are defined as properties